### DISCUSSION TITLE
Permissions for Admins and / or Multiple roles

### DISCUSSION THREAD

**Discussion by jeremybradbury** [directus/directus#12473] [Feature Requests]
Not just specifically using system tables (but that's my use case), Admins cannot be given more or less access to anything. 

A related example: we cannot allow our customer service team to edit or correct `external_identifier`s or allow users to see their own. We can't even let them see those fields without an Admin role (CS or the User).

Other roles cannot have access to settings, not without full admin permissions.

This creates a confusing and unsolvable permissions matrix.

A fix would be to add a permission entry when needed for admins, with the ability to reset to default.

Another fix would be more widely beneficial: make roles O2M instead of O2O.

If I could add a role to some admins to add/remove some permissions, we wouldn't need admin permissions, just role priority or sorting like Discord.

For example: default `Admin`s can view settings, add & edit fields but not delete tables or relations. I add the `superuser` role and I can wipe all the things. Or I add the `readonly` role and we can restrict all edits.

This would also allow end users to have say 3 roles each allowing access to different data sets and possibly being reflected in the user profile, again like Discord's role system.

My claim: 

I should be able to have a user that's a `partner`, `streamer`, `employee`, `player` and an `admin`. they should have a priority/sort order given for the list, then another order for each user. so likely a pivot table with a sort number attached to each pair, in addition to sort on the roles table (another key missing feature, sorting meta for M2Ms where you can drag to sort).

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
